File: Cpp/1637_Widest_Vertical_Area_Between_Two_Points_Containing_No_Points.cpp
class Solution {
public:
    int maxWidthOfVerticalArea(vector<vector<int>>& points) {
        vector<int> v;
        int result = 0;
        for(vector<int> point : points){
            v.push_back(point[0]);
        }
        sort(v.begin() ,v.end());
        //quicksort
        //quicksort(v, 0, v.size()-1);
        
        for(int i = 1; i < v.size(); i++){
            result = max(result, v[i] - v[i-1]);
        }
        
        return result;
    }
    /*
    void quicksort(vector<int> &v, int left, int right){
        if(left >= right){
            return;
        }
        
        int key = v[left];
        int i = left;
        int j = right;
        while(i != j){
            while(v[j] > key && i < j){
                j--;
            }
            while(v[i] <= key && i < j){
                i++;
            }
            if(j > i){
                int temp = v[i];
                v[i] = v[j];
                v[j] = temp;
            }
        }
        v[left] = v[i];
        v[i] = key;
        quicksort(v, left, i-1);
        quicksort(v, i+1, right);
    }
    */
};

File: CS/1637_Widest_Vertical_Area_Between_Two_Points_Containing_No_Points.cs
// 2022-03-07 14:58~15:09

public class Solution {
    public int MaxWidthOfVerticalArea(int[][] points) {
        List<int> x = new List<int>();
        for(int i = 0; i < points.Length; i++){
            x.Add(points[i][0]);
        }
        x.Sort();
        int result = 0;
        for(int i = 1; i < x.Count; i++){
            result = Math.Max(result, x[i] - x[i-1]);
        }
        return result;
    }
}
